server/routes/incidentDetails/incidentDetailsPage.ts

```diff
@@ -0,0 +1,728 @@
+      ),
+      exitButtonData,
+      formData: transformStashedDataToFormData(data, originalReporterUsername),
+    }
+  }
+
+  getPageDataOnPost = async (postValues: SubmittedFormData, currentUser: User): Promise<PageData> => {
+    let exitButtonData: ExitButtonData = null
+    if (this.pageOptions.isEdit()) {
+      let prisonerReportUrl = null
+      if (this.pageOptions.isPreviouslySubmitted()) {
+        prisonerReportUrl = postValues.originalPageReferrerUrl
+      }
+      exitButtonData = {
+        prisonerNumber: postValues.prisonerNumber,
+        draftId: postValues.draftId,
+        prisonerReportUrl,
+      }
+    }
+    return {
+      displayData: await this.getDisplayData(
+        postValues.prisonerNumber,
+        postValues.originalReporterUsername,
+        currentUser,
+        postValues.incidentDetails?.currentAssociatedPrisonerNumber
+      ),
+      exitButtonData,
+      formData: postValues,
+    }
+  }
+
+  getDisplayData = async (
+    prisonerNumber: string,
+    originalReporterUsername: string,
+    currentUser: User,
+    currentAssociatedPrisonerNumber?: string
+  ): Promise<DisplayData> => {
+    const [prisoner, reporter] = await Promise.all([
+      this.placeOnReportService.getPrisonerDetails(prisonerNumber, currentUser),
+      this.placeOnReportService.getReporterName(originalReporterUsername, currentUser),
+    ])
+    const { agencyId } = prisoner.assignedLivingUnit
+    const possibleLocations = await this.locationService.getIncidentLocations(agencyId, currentUser)
+
+    const associatedPrisonersName = currentAssociatedPrisonerNumber
+      ? await this.getCurrentAssociatedPrisonersName(currentAssociatedPrisonerNumber, currentUser)
+      : null
+
+    return {
+      reporter,
+      prisoner,
+      possibleLocations,
+      associatedPrisonersName,
+    }
+  }
+
+  getCurrentAssociatedPrisonersName = async (associatedPrisonersNumber: string, user: User) => {
+    if (!associatedPrisonersNumber) return null
+    const associatedPrisoner = await this.placeOnReportService.getPrisonerDetails(associatedPrisonersNumber, user)
+    return associatedPrisoner.displayName
+  }
+}
+
+const extractValuesFromRequest = (req: Request): RequestValues => {
+  const values = {
+    draftId: getDraftIdFromString(req.params.id),
+    prisonerNumber: req.params.prisonerNumber,
+    selectedPerson: JSON.stringify(req.query.selectedPerson)?.replace(/"/g, ''),
+    deleteWanted: req.query.personDeleted as string,
+    originalPageReferrerUrl: req.query.referrer as string,
+  }
+  return values
+}
+
+const getPageActionFromRequest = (reqData: RequestValues): PageRequestAction => {
+  if (reqData.selectedPerson) {
+    return PageRequestAction.SEARCH_FOR_PRISONER
+  }
+  if (reqData.deleteWanted) {
+    return PageRequestAction.DELETE_PRISONER
+  }
+  return PageRequestAction.STANDARD
+}
+
+const popDataFromSession = (req: Request): StashedIncidentDetails => {
+  const { incidentDate } = req.session
```

Comment:
Maybe for another time - but is it worth having an object on the session instead of distinct properties?

Reply:
Ha yeah - that does make sense!